Add links to contributed custom scalar specs at scalars.graphql.org (#1009)
graphql/graphql-spec: spec/Section 3 -- Type System.md

:: When defining a custom scalar, GraphQL services should provide a _scalar
specification URL_ via the `@specifiedBy` directive or the `specifiedByURL`
introspection field. This URL must link to a human-readable specification of the

> data format, serialization, and coercion rules for the scalar. See more on the GraphQL scalars [implementation guide](https://scalars.graphql.org/implementation-guide) and see [custom scalar specification templates](https://scalars.graphql.org/) at [scalars.graphql.org](https://scalars.graphql.org/).

Reviewer: On second thoughts, since we have this detail below, I don't think we need any edits here. Suggest we merge this into the comment on line 415ish below.

specified by this document.

Note: Custom scalars should also summarize the specified format and provide

> examples in their description. See the GraphQL scalars [implementation guide](https://scalars.graphql.org/implementation-guide) and see [custom scalar specification templates](https://scalars.graphql.org/) at [scalars.graphql.org](https://scalars.graphql.org/).

Reviewer: Let's limit it to just one link; the implementation guide itself should link out to other resources if necessary.

```suggestion
examples in their description; see the GraphQL scalars [implementation guide](https://scalars.graphql.org/implementation-guide) for more guidance.
```

definition language to provide a _scalar specification URL_ for specifying the
behavior of [custom scalar types](#sec-Scalars.Custom-Scalars). The URL should
point to a human-readable specification of the data format, serialization, and

> coercion rules. See more on the GraphQL scalars [implementation guide](https://scalars.graphql.org/implementation-guide) and see [custom scalar specification templates](https://scalars.graphql.org/) at [scalars.graphql.org](https://scalars.graphql.org/). This directive must not appear on built-in scalar types.
> 
> You can host your custom scalar specification on the GraphQL Foundation owned domain [scalars.graphql.org](https://scalars.graphql.org/), see how to add your specification on the [contribution guide](https://scalars.graphql.org/readme-contribution-guide).

Reviewer: Let's move this out of the normative section, and reduce it to the one link:

```suggestion
coercion rules. It must not appear on built-in scalar types.

Note: Details on implementing a GraphQL scalar specification can be found in the [scalars.graphql.org implementation guide](https://scalars.graphql.org/implementation-guide).
```

point to a human-readable specification of the data format, serialization, and
coercion rules. It must not appear on built-in scalar types.

Note: Details on implementing a GraphQL scalar specification can be found in the [scalars.graphql.org implementation guide](https://scalars.graphql.org/implementation-guide).

In this example, a custom scalar type for `UUID` is defined with a URL pointing
to the relevant IETF specification.

```graphql example
scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")

> scalar URL @specifiedBy(url: "https://tools.ietf.org/html/rfc3986")

Reviewer 2: I honestly think these are not good examples: we should not encourage people to link directly 
to RFCs. I would even suggest to remove the UUID example. A RFC is great to be the basis of a custom scalar, but it is almost never sufficient to be a good guide. We should people actively encourage to write dedicated custom scalars specs.

Reviewer: Agreed; writing a scalars.graphql.org spec that calls out to the relevant RFC seems wise. I'm not sure if that needs to be done in this PR though?

Reviewer: It definitely doesn't need to be addressed in this PR; it's already referenced on line 407, this is just a copy. Can someone file it for follow-up though?

> scalar URL @specifiedBy(url: "https://tools.ietf.org/html/rfc3986")
> scalar DateTime @specifiedBy(url: "https://scalars.graphql.org/andimarek/date-time")

Reviewer: The text before this block says "In this example, a custom scalar type for UUID is defined with a URL pointing to the relevant IETF specification." therefore the block should only define UUID. The prior examples are sufficient, no need for change here.

```suggestion
```
